### PR TITLE
Add script for downloading triage logs

### PR DESCRIPTION
**Pull request by empovit**

### CODE REVIEW

**Review by javipolo — APPROVED** [review]

File: triage/triage-scripts/download-logs.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env python3
+
+# This script takes a URL of a directory in the NGINX-file browser format (https://github.com/mohamnag/nginx-file-browser),
+# and downloads its contents into a subdirectory inside the working directory. A name of the destination directory can be
+# either given as the second argument (after the URL), or will be derived from the last part of the URL. Files in the known
+# archive formats will be unpacked. The subdirectory will be relative to the working directory.
+#
+# Example: http://logs.example.com/#/2022-03-30_20:14:19_fab47716-04e6-401b-8b89-e414d030a298/ will be converted into
+# http://logs.example.com/files/2022-03-30_20:14:19_fab47716-04e6-401b-8b89-e414d030a298/, which returns a JSON list of files
+# and subdirectories. The files/directories will be recursively downloaded into "2022-03-30_20-14-19_fab47716-04e6-401b-8b89-e414d030a298".
```
Comment (osherdp) [inline triage/triage-scripts/download-logs.py:10]:
```suggestion
"""
This script takes a URL of a directory in the NGINX-file browser format (https://github.com/mohamnag/nginx-file-browser),
and downloads its contents into a subdirectory inside the working directory. A name of the destination directory can be
either given as the second argument (after the URL), or will be derived from the last part of the URL. Files in the known
archive formats will be unpacked. The subdirectory will be relative to the working directory.

Example: http://logs.example.com/#/2022-03-30_20:14:19_fab47716-04e6-401b-8b89-e414d030a298/ will be converted into
http://logs.example.com/files/2022-03-30_20:14:19_fab47716-04e6-401b-8b89-e414d030a298/, which returns a JSON list of files
and subdirectories. The files/directories will be recursively downloaded into "2022-03-30_20-14-19_fab47716-04e6-401b-8b89-e414d030a298".
"""
```

**Review by osherdp — CHANGES_REQUESTED** [review]
Maybe just change dirs hierarchy, because it's a bit redundant to have ``assisted-scripts/triage/triage-scripts/``
Something like ``triage/triage-scripts/ -> triage/logs-gathering/``

File: triage/triage-scripts/download-logs.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        if f_type == "file":
+            download_file(path.join(base_url, f_name), path.join(dest, f_name))
+            try_unpack(dest, f_name)
+        elif f_type == "directory":
+            download_dir(path.join(base_url, f_name), path.join(dest, f_name))
+        else:
+            print(f"Unknown type: {f_type}")
+
+def download_file(base_url, dest):
+    r = requests.get(base_url, stream=True)
```
[the remark this answers is not in the document]
Comment (osherdp) [reply triage/triage-scripts/download-logs.py:37]: otherwise you might download content of 4xx/5xx page without knowing it

**Comment by javipolo** [comment]
> Maybe just change dirs hierarchy, because it's a bit redundant to have `assisted-scripts/triage/triage-scripts/` Something like `triage/triage-scripts/ -> triage/logs-gathering/`

I would make a catch-all directory for those small scripts, like assisted-scripts/triage/scripts or something like that

I doubt there will be much log-gathering extra things, and having a directory per script seems a little bit too much

**Comment by osherdp** [comment]
> > Maybe just change dirs hierarchy, because it's a bit redundant to have `assisted-scripts/triage/triage-scripts/` Something like `triage/triage-scripts/ -> triage/logs-gathering/`
> 
> I would make a catch-all directory for those small scripts, like assisted-scripts/triage/scripts or something like that
> 
> I doubt there will be much log-gathering extra things, and having a directory per script seems a little bit too much

So in this case just putting the new script directly under triage/. This whole repo is scripts, so no need to remind of this content in any subdirectory 😄

**Comment by empovit** [comment]
PTAL

**Review by osherdp — APPROVED** [review]

**Pull request merged**